heka/_service.sls

```diff
@@ -52,61 +50,68 @@ heka_{{ service_name }}_service_wrapper:
   - source: salt://heka/files/service_wrapper
   - user: root
   - mode: 755
+  - defaults:
+    service_name: {{ service_name }}
   - template: jinja
 
 {%- endif %}
 
+heka_{{ service_name }}_service:
+  service.running:
+  - name: {{ service_name }}
+  - enable: True
 
 {# Setup basic structure for all roles so updates can apply #}
 
 {%- set service_grains = {
-  'heka': {
-    'log_collector': {
-      'decoder': {},
-      'input': {},
-      'filter': {},
-      'splitter': {},
-      'encoder': {},
-      'output': {}
-    },
-    'metric_collector': {
-      'decoder': {},
-      'input': {},
-      'filter': {},
-      'splitter': {},
-      'encoder': {},
-      'output': {}
-    },
-    'remote_collector': {
-      'decoder': {},
-      'input': {},
-      'filter': {},
-      'splitter': {},
-      'encoder': {},
-      'output': {}
-    },
-    'aggregator': {
-      'decoder': {},
-      'input': {},
-      'filter': {},
-      'splitter': {},
-      'encoder': {},
-      'output': {}
-    }
+  'log_collector': {
+    'decoder': {},
+    'input': {},
+    'filter': {},
+    'splitter': {},
+    'encoder': {},
+    'output': {}
+  },
+  'metric_collector': {
+    'decoder': {},
+    'input': {},
+    'filter': {},
+    'splitter': {},
+    'encoder': {},
+    'output': {}
+  },
+  'remote_collector': {
+    'decoder': {},
+    'input': {},
+    'filter': {},
+    'splitter': {},
+    'encoder': {},
+    'output': {}
+  },
+  'aggregator': {
+    'decoder': {},
+    'input': {},
+    'filter': {},
+    'splitter': {},
+    'encoder': {},
+    'output': {}
   }
 } %}
 
 
+
 {# Loading the other services' support metadata for local roles #}
 
 {%- if service_name in ['log_collector', 'metric_collector'] %}
 
 {%- for service_name, service in pillar.iteritems() %}
 {%- if service.get('_support', {}).get('heka', {}).get('enabled', False) %}
 
+{%- macro load_grains_file(grains_fragment_file) %}{% include grains_fragment_file %}{% endmacro %}
```

Comment:
I think you need to use `ignore missing` here, otherwise it will fail with for example the `salt` service which does support `heka` but does not have an `heka.yml` meta file.

http://jinja.pocoo.org/docs/dev/templates/#include